lib/formz.dart

```diff
@@ -148,8 +150,16 @@ mixin FormzInputErrorCacheMixin<T, E> on FormzInput<T, E> {
 class Formz {
   /// Returns a [bool] given a list of [FormzInput] indicating whether
   /// the inputs are all valid.
-  static bool validate(List<FormzInput<dynamic, dynamic>> inputs) {
-    return inputs.every((input) => input.isValid);
+  static FutureOr<bool> validate(
+    List<FormzInput<dynamic, dynamic>> inputs,
+  ) async {
+    var allValid = true;
+    for (final input in inputs) {
+      if (await input.isNotValid) {
+        allValid = false;
+      }
+    }
```

Comment:
We should terminate as soon as one is not valid. I think there is no need to keep checking since we are not collecting errors.

Reply:
I did the change you ask and use the NameInputErrorCacheMixin helper to write the test, let me know what you think :)

Reply:
I made some amendments to avoid an scenario where we resolve as soon as we know an input is false. 

For example, given two inputs: one with an async validator that resolves to false in 1 second and another input that resolves to true in 5 minutes, we shouldn't wait 5 minutes to check if the form is valid since you already know that it is not.